controllers/kuadrant_controller.go

```diff
@@ -389,14 +396,47 @@ func (r *KuadrantReconciler) createOnlyInKuadrantNSCb(ctx context.Context, kObj
 			return err
 		}
 
-		k8sObjKind := k8sObj.DeepCopyObject().GetObjectKind()
+		var newObj client.Object
+		newObj = k8sObj
```

Comment:
I must be missing something here, but there is a [lot](https://github.com/Kuadrant/kuadrant-operator/pull/22/files#diff-53e20feafdf0308bc6f15e87b2dc63acd05e732a6bf70bc5247745edd59b1e4cR409) [of](53e20feafdf0308bc6f15e87b2dc63acd05e732a6bf70bc5247745edd59b1e4cR419) [copying](53e20feafdf0308bc6f15e87b2dc63acd05e732a6bf70bc5247745edd59b1e4cR419) going on here. 
Is the `switch obj := k8sObj.(type)` a copy itself? Trying to understand… but I wonder why not use the `default:` if a "default" copy has to happen? That's if they are all different… I guess I need to refresh my golang

Reply:
I know, it's a bit misleading the `default` case, its block will be executed if none of the other cases match.

Reply:
`switch obj := k8sObj.(type)` will only live within the switch block, so we need to copy it to a different `var`. 

I'd love the swith case to include a `finally` clause, that will be executed at the very end, thus we could avoid a lot of repetition